canonical-data.schema.json

```diff
@@ -1,166 +1,159 @@
-                , "scenarios"   : { "$ref": "#/definitions/scenarios"   }
-                , "comments"    : { "$ref": "#/definitions/comments"    }
-                , "property"    : { "$ref": "#/definitions/property"    }
-                , "input"       : { "$ref": "#/definitions/input"       }
-                , "expected"    : { "$ref": "#/definitions/expected"    }
-                }
-          , "additionalProperties": false
-          },
-
-      "labeledTestGroup":
-          { "description": "A group of tests with a description"
-          , "type"       : "object"
-          , "required"   : ["description", "cases"]
-          , "properties" :
-                { "description": { "$ref": "#/definitions/description" }
-                , "scenarios"  : { "$ref": "#/definitions/scenarios"   }
-                , "comments"   : { "$ref": "#/definitions/comments"    }
-                , "cases"      : { "$ref": "#/definitions/testGroup"   }
-                }
-          , "additionalProperties": false
-          },
-
-      "description":
-          { "description": "A short, clear, one-line description"
-          , "type"       : "string"
-          },
-
-      "uuid":
-          { "description": "A version 4 UUID (compliant with RFC 4122) in the canonical textual representation"
-          , "type"       : "string"
-          , "pattern"    : "^[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$"
-          },
-
-      "scenario":
-          { "description": "An identifier for a specific scenario (kebab-case)"
-          , "type": "string"
-          , "enum":
-                [ "big-integer"
-                , "floating-point"
-                , "immutable"
-                , "library-test"
-                , "local-scope"
-                , "unicode"
-                , "input-validation"
-                , "runtime-validation"
-                ]
-          },
-
-      "scenarios":
-          { "description": "An array of scenarios that can be used to include/exclude test cases"
-          , "type"       : "array"
-          , "items"      : { "$ref": "#/definitions/scenario" }
-          , "minItems"   : 0
-          },
-
-      "property":
-          { "description": "A letters-only, lowerCamelCase property name"
-          , "type"       : "string"
-          , "pattern"    : "^[a-z]+([A-Z][a-z]+)*[A-Z]?$"
-          },
-
-      "input":
-          { "description": "The inputs to a test case"
-          , "type"       : "object"
-          },
-
-      "expected":
-          { "description": "The expected return value of a test case"
-          , "properties":
-               { "error": { "$ref": "#/definitions/error" }
-               }
-          , "dependencies":
-               { "error": { "maxProperties": 1 }
-               }
-          },
-
-      "error":
-          { "description": "A message describing an error condition"
-          , "type"       : "string"
-          }
-
+  "comments": [
```

Comment:
The schema previously used a strange commas-first format. I don't particularly care either way but I assume it was done that way for a reason.

Reply:
Perhaps, maybe someone can remember? To me it always looked very odd and it was widely different from the other files.

Reply:
No specific reason is given in https://github.com/exercism/problem-specifications/pull/602. That's just how all JSON in that PR was formatted. It does not appear to me to have any significance other than that some decision needed to be made for how to format the file, and that particular choice, equally as valid as any other, was chosen.

Reply:
In that case it's probably best to format it to the standard JSON format as well :)